test/chrome/crdpMultiplexing/crdpMultiplexor.test.ts

```diff
@@ -0,0 +1,175 @@
+        webSocketMock = undefined;
+        socketMessageCallbacks = [];
+    });
+
+    test('Channel is successfully added', done => {
+        let testChannel = multiplexor.addChannel("testChannel");
+        Assert.notEqual(testChannel, null, "Test channel should not be null");
+        done();
+    });
+
+    test('Multiplexor message sent to underlying socket and response received.', done => {
+        let channel = multiplexor.addChannel("testChannel");
+        let messageId = 1;
+        let channelCallback = (data: string) => {
+            // Test receiving data - called when socket sends back response data
+            Assert.equal(JSON.parse(data).id, messageId);
+            done();
+        };
+
+        // Test sending data
+        channel.on("message", channelCallback);
+        multiplexor.send(channel, '{"method":"Runtime.enable","id":' + messageId + '}');
+        webSocketMock.verify(s => s.send(It.isAnyString()), Times.atLeastOnce());
+    });
+
+    test('Multiplexor message handles multiple channels', () => {
+        let callbackPromises: Promise<void>[] = [];
+
+        for (let i = 1; i < 5; i++) {
+            let channel = multiplexor.addChannel("Channel" + i);
+
+            let callbackPromise = new Promise<void>(function (resolve, reject) {
+                let callback = (data: string) => {
+                    Assert.equal(JSON.parse(data).id, i);
+                    resolve();
+                }
+
+                channel.on("message", callback);
+            });
+
+            callbackPromises.push(callbackPromise);
+            multiplexor.send(channel, '{"method":"Console.enable","id":' + i + '}');
+        }
+
+        // wait for all promises
+        return Promise.all(callbackPromises);
+    });
+
+    test('Multiplexor notification delivered to all channels', () => {
+        let callbackPromises: Promise<void>[] = [];
+        let testNotification: string = "Test.notification";
+        let testEnable: string = "Test.enable";
+
+        for (let i = 1; i < 5; i++) {
+            let channel = multiplexor.addChannel("Channel" + i);
+
+            let callbackPromise = new Promise<void>(function (resolve, reject) {
+                let callback = (data: string) => {
+                    let notification = JSON.parse(data);
+                    if (notification.id !== undefined) {
+                        // first callback is the response to test.enable
+                        Assert.equal(notification.id, i);
+                    } else {
+                        // second callback is the Test.notification callback
+                        Assert.equal(notification.method, testNotification);
+                        resolve();
+                    }
+                }
+
+                channel.on("message", callback);
+            });
+
+            // Enable the Test domain so we will receive notifications
+            channel.send('{"method":"' + testEnable + '","id":' + i + '}');
```

Comment:
Even though we are enabling the domains in sync way now, we might convert it to async in the future. It'd be great if the same test still worked...